src/debugger.ts

```diff
@@ -121,41 +121,44 @@ export class Debugger
     this.subscriptions.forEach((subscription) => subscription.dispose());
   }
 
+  private getSockets(): string[] {
+    const cmd = "bundle exec rdbg --util=list-socks";
+    let sockets: string[] = [];
+    try {
+      sockets = execSync(cmd)
+        .toString()
+        .split("\n")
+        .filter((socket) => socket.length > 0);
+    } catch (error: any) {
+      this.console.append(error.message);
+    }
+    return sockets;
```

Comment:
Let's import `asyncExec` from common instead of using synchronous exec.

```suggestion
    try {
      const result = asyncExec(cmd, { cwd: this.workingFolder, env: this.ruby.env });
      
      return result.stdout
        .toString()
        .split("\n")
        .filter((socket) => socket.length > 0);
    } catch (error: any) {
      this.console.append(`Error listing sockets: ${error.message}`);
    }
```

Reply:
I added the cwd and env and applied suggested error handling improvement, unfortunately I can't find asyncExec in common or anywhere else actually. Pls point me to what exactly you mean.

child_process contains asynchronous `exec`. This could be used, though we're waiting for the result anyway as it's to be used immediately. So I'm thinking there would be no real benefit from complicating the implementation.

The docs say:
The child_process.execSync() method is generally identical to [child_process.exec()](https://nodejs.org/docs/latest-v16.x/api/child_process.html#child_processexeccommand-options-callback) with the exception that the method will not return until the child process has fully closed.